sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -669,17 +669,26 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
     )
   }
 
+  private def getAliasFunc(alias: => String): Option[Expression => String] = {
+    if (conf.getConf(SQLConf.STABLE_DERIVED_COLUMN_ALIAS_ENABLED)) {
+      Some(_ => alias)
+    } else {
+      None
+    }
+  }
+
   override def visitNamedExpressionSeq(
-      ctx: NamedExpressionSeqContext): Seq[Expression] = {
+      ctx: NamedExpressionSeqContext): Seq[(Expression, Option[Expression => String])] = {
     Option(ctx).toSeq
       .flatMap(_.namedExpression.asScala)
-      .map(typedVisit[Expression])
+      .map(ctx => (typedVisit[Expression](ctx), getAliasFunc(toExprAlias(ctx))))
```

Comment:
it seems we can always create `UnresolvedAlias` here. `UnresolvedAlias` is a noop if its child is already a named expression.

Reply:
Just in case, this function handle not only named expressions but others too. For example:
```sql
select 1;
```
is handled by `visitNamedExpressionSeq()` but not by `visitExpressionSeq()`.